src/NodeValueWriter.cpp

```diff
@@ -247,13 +280,41 @@ void NodeValueWriter::write_wstring(const ACE_CDR::WChar* value, size_t length)
   v8::MaybeLocal<v8::String> v8_value = Nan::New<v8::String>(str, length);
   value_helper<v8::String>(v8_value);
   delete [] str;
+  return true;
 }
 
-void NodeValueWriter::write_enum(const char* name, ACE_CDR::Long value)
+bool NodeValueWriter::write_enum(ACE_CDR::Long value, const OpenDDS::DCPS::EnumHelper& helper)
 {
-  ACE_UNUSED_ARG(value);
+  const char* name = 0;
+  if (!helper.get_name(name, value)) {
+    return false;
+  }
   v8::MaybeLocal<v8::String> v8_value = Nan::New<v8::String>(name);
   value_helper<v8::String>(v8_value);
+  return true;
+}
+
+bool NodeValueWriter::write_bitmask(ACE_CDR::ULongLong value, const OpenDDS::DCPS::BitmaskHelper& helper)
+{
+  const OpenDDS::DCPS::String flags = bitmask_to_string(value, helper);
```

Comment:
This is to convert the bitmask to a string formed by the flag names separated by the `|` character. This string is used for the write instead of a number representation.

Reply:
It seems like it would be better as an sequence of flag names or even better for javascript, a plain object where each flag has a true or false value. That would make it easier to test if the bitmask had something enabled. With a single string the user would have to process it to find out what it has.

Reply:
I don't think the bit positions are all that important, but being able to actually get the boolean values would be entire point of the bitmask.

Reply:
Even in JSON, I'm not sure that format is optimal now that I think about it. If you want to push the JSON through a tool like jq and filter based on a bitmask flag, it would be easier if the flag values were explicit.